src/aspire/denoise/covar2d.py

```diff
@@ -33,12 +45,60 @@ def get_covar(self, coeffs=None, mean_coeff=None,  do_refl=false):
         :return: The covariance matrix of coefficients for all images.
 
         """
-        pass
+        if coeffs is None:
+            raise RuntimeError('The coefficients need to be calculated!')
+        if mean_coeff is None:
+            mean_coeff = self.get_mean(coeffs)
+
+        covar_coeff = []
+        ind = 0
+        ell = 0
+        mask =  self.basis._indices["ells"] == ell
+        coeff_ell = coeffs[mask, ...] - mean_coeff[mask, ...]
+        # covar_ell = np.zeros((coeff_ell.shape[0], coeff_ell.shape[0]))
+        covar_ell = np.array(coeff_ell @ coeff_ell.T/np.size(coeffs, 1))
+        #print("covar_ell", np.size(covar_ell))
+        covar_coeff.append(covar_ell)
+        ind += 1
+
+        for ell in range(1, self.basis.ell_max+1):
+            mask = self.basis._indices["ells"] == ell
+            #mask_pos = mask and (self.basis._indices['sgns'] == +1)
```

Comment:
Can we kill all commented out code? Here as well as anywhere else in the PR.

Reply:
They are cleared up in the new commit.